### PR TITLE
Fixes #1945 Restore library panels

### LINKED ISSUE #1945: Save/restore library panels when the app is reopened
**Is your feature request related to a problem? Please describe.**
At the moment FxR is saving the layout and restores it after the app is reopened but in only restores browser windows, not library panels.

**Describe the solution you'd like**
I'd like to be able to restore also library panels so my exact layout is restored upon app reopen. See the screenshot.

**Describe alternatives you've considered**
Also would be great if I could save/restore specific layouts that I might use for different tasks. For example a layout that I use for reading different newspapers, a layout for developing or a layout for general browsing or watching movies.

**Additional context**
org mozilla vrbrowser dev-20191008-234253

### PR DESCRIPTION
**Pull request by keianhzo**
Fixes #1945 Restore library panels. I really don't like to have a flag to know if the user has initiated the interaction so if there is a better alternative I'd be happy to change this. Without the flag the panels would be closed during the restore when about:blank and home are loaded as we always close them when there is a navigation event.

### CODE REVIEW

File: app/src/common/shared/org/mozilla/vrbrowser/ui/widgets/Windows.java [hunk 1]
```diff
@@ -275,6 +289,14 @@ private WindowWidget addRestoredWindow(@NonNull WindowState aState, @NonNull Ses
         newWindow.getPlacement().worldWidth = aState.worldWidth;
         newWindow.setRestored(true);
         placeWindow(newWindow, aState.placement);
+        switch (aState.panelType) {
```
Comment (bluemarvin) [inline app/src/common/shared/org/mozilla/vrbrowser/ui/widgets/Windows.java]: I'm getting a crash here:
https://crash-stats.mozilla.com/report/index/ded91d9d-9f44-46ba-bc9c-18fca0200123
https://crash-stats.mozilla.com/report/index/e8e5b8d8-e61a-4b3a-87e9-9770f0200123
STR:
1) Run FxR without this patch applied and quit so windows are saved.
2) Apply patch and re-run FxR
FxR will crash on start up.

Comment (keianhzo) [reply app/src/common/shared/org/mozilla/vrbrowser/ui/widgets/Windows.java]: Good catch. I've added a check to see if the new panel information is present.

**Comment by keianhzo** [comment]
This should be ready for review.

**Comment by bluemarvin** [comment]
I agree this isn't pretty but I can't think of anything cleaner. @MortimerGoro last chance to comment before landing it.

**Comment by MortimerGoro** [comment]
@bluemarvin @keianhzo would it be possible to use a URL based approach instead of adding the extra flags and enum? 

For example map `about://history` or `about://bookmarks` kind of URLs to the library panels

**Comment by bluemarvin** [comment]
> @bluemarvin @keianhzo would it be possible to use a URL based approach instead of adding the extra flags and enum?
> 
> For example map `about://history` or `about://bookmarks` kind of URLs to the library panels

That could be workable. I'm not sure how we would make it work with the GV history system though. I like the concept, it isn't clear how we would implement it.

**Comment by keianhzo** [comment]
@bluemarvin @MortimerGoro That a good idea, I've added support for it. It doesn't seem to mess up the history and it's a much cleaner solution for restoring.

**Pull request merged**